### PR TITLE
Adds a code of conduct

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
Requested in #2891

You can see the latest version of the PR here https://github.com/holoviz/panel/blob/code-of-conduct/CODE_OF_CONDUCT.md.

[image](https://github.com/holoviz/panel/blob/code-of-conduct/CODE_OF_CONDUCT.md)

# Todo

- [] Identify and appoint community leaders responsible for (enforcing) the code of conduct.
- [] Take the time to discuss enforcement with trusted members of the community. 
- [] Thoroughly read the Enforcement Guidelines section of the document and adapt if necessary
- [] Assign *code of conduct* email.
- [] Add a link to this document in the [HoloViz Discourse FAQ](https://discourse.holoviz.org/faq)

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+# Contributor Covenant Code of Conduct
+
+## 👩🏽‍👩🏽‍👦🏽‍👧🏽 Summary
+
+The Panel community is fantastic. We wan't it to stay that way.
```
Comment (jbednar) [inline CODE_OF_CONDUCT.md:5]:
```suggestion
The Panel community is fantastic. We want it to stay that way.
```

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+an individual is officially representing the community in public spaces.
+Examples of representing our community include using an official e-mail address,
+posting via an official social media account, or acting as an appointed
+representative at an online or offline event.
+
+## 👩‍⚖️ Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the community leaders responsible for enforcement at
+[INSERT CONTACT METHOD].
```
Comment (jbednar) [inline CODE_OF_CONDUCT.md:71]: Any suggestions for an appropriate contact method?

**Review by jbednar — COMMENTED** [review]
Thanks for kickstarting this! Should we maybe do this at the HoloViz.org level rather than Panel? Many issues and discussions cross multiple github repositories and share the common discourse server, wiki, and other forums, so having separate codes of conduct in those different spaces seems unwieldy. Of course, the code should be prominently linked from each site in any case.

**Comment by MarcSkovMadsen** [comment]
Thanks @jbednar . I was thinking the same. I think we should take the time to discuss and set this up. But I also think its important for a project like HoloViz to have a code of conduct. Both as a signal but also the just in case for the day when its needed.

One thought I had was that maybe we could find users interested in shaping/ steering this effort?

File: CODE_OF_CONDUCT.md (hunk 2 above)
Comment (MarcSkovMadsen) [reply CODE_OF_CONDUCT.md:71]: coc@holoviz.org or something similar. But I think it requires forming a code of conduct team/ council.

**Comment by philippjfr** [comment]
Very much in favor of this, so thanks for getting this started. I'd like to consult with a few other projects on what the best template for a CoC is before deciding to go with this particular one. Also a strong +1 to ask around in the community if anyone is interested in joining this effort.

**Comment by jbednar** [comment]
I'm happy with the Pandas approach to the [COC](https://github.com/pandas-dev/pandas/blob/master/.github/CODE_OF_CONDUCT.md), which is to have a mailing list pandas-coc@googlegroups.com and a public declaration of who is on that list. I'd go one further and add a sentence "If your complaint involves anyone who is on this list, please reach out separately to one or more members of the list who have not been involved in the behavior in question". Otherwise I don't see how to raise an issue privately if it involves someone on the COC mailing list.

**Comment by philippjfr** [comment]
Working on setting up the CoC email address, for now I'd suggest @MarcSkovMadsen, @jbednar and I to be on that list. Will also ask around on the Discourse if anyone else is interested in joining the CoC committee.

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
```
Comment (philippjfr) [inline CODE_OF_CONDUCT.md:71]:
```suggestion
coc@holoviz.org.
```

**Comment by philippjfr** [comment]
@MarcSkovMadsen @jbednar and @philippjfr are now on the CoC committee. Future appointees will be discussed at HoloViz meetings, which we will open to the public going forward.

**Comment by philippjfr** [comment]
Thanks @MarcSkovMadsen for getting the ball rolling.

**Pull request merged**